test/lib/sync-task.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+var expect = require('expect');
+var Semver = require('sver-compat').Semver;
+var eraseTime = require('gulp-test-tools').eraseTime;
+var runner = require('gulp-test-tools').gulpRunner;
+
+
+function nodeDoesNotNodeSupportExitCode() {
+  var currentVersion = new Semver(process.versions.node);
+  var minimalVersion = new Semver('0.11.0');
+  return currentVersion.lt(minimalVersion);
+}
+
+describe('sync-task', function() {
+  it('should return error code 1 if any tasks did not complete', function(done) {
+    var this_ = this;
```

Comment:
@barbogast It's not necessary to use the new package. As @phated said, this test case can be skipped by the following code at this point:

```js
  it('should return error code 1 if any tasks did not complete', function(done) {
    if (!('exitCode' in process)) { // On v0.10 only, this condition is true because process doesn't have .exitCode.
      this.skip();
      return;  // This is not necessary because skip() throws an exception, but is recommened to avoid confusing.
    }
    ...
```

Reply:
Are you sure this will work? I just rechecked and 

```
> 'exitCode' in process
false
```
for 
```
node -v
v8.9.3
```

Or am I misunderstanding completely?

Reply:
Ah, I think I understood. Do you mean that `exitCode` was set already by the new line in `sync-task.js`? This would work, but only if `require('gulp-test-tools').gulpRunner` which is used by the test doesn't spawn a new process, right? I guess that it does but I'll try when I get home.

Reply:
@barbogast I'm sorry, I really misunderstood. You are right. I did only checked process.exitCode is undefined only on v0.10 and I completely thought process.exitCode is zero on other versions.

Reply:
@barbogast Since these test cases in `test/lib/sync-task.js` are not unit test, please move this file to `test/` and rename it to another like `exit-code.js`.

In addition, I think it's better that these two cases can be put into one case and then `nodeDoesNotNodeSupportExitCode()` is used not for `this.skip()` but for unchecking exit code.

Reply:
Please move the file like sttk mentioned and remove that utility (using my other recommendation above).